lib/protocol/protocol.go

```diff
@@ -237,8 +243,9 @@ func NewConnection(deviceID DeviceID, reader io.Reader, writer io.Writer, receiv
 // Start creates the goroutines for sending and receiving of messages. It must
 // be called exactly once after creating a connection.
 func (c *rawConnection) Start() {
+	go c.readerLoop()
 	go func() {
-		err := c.readerLoop()
+		err := c.receiverLoop()
 		c.internalClose(err)
```

Comment:
I guess this will result in a double close, but it's a sync.Once so it's ok.
The alternative would be to call internalClose with within the receiver, and not call internalClose on <-c.close.
But it doesn't matter much I guess.

Reply:
>  But it doesn't matter much I guess.

That. And that I don't want to touch this code unless I have to and it's tested (and I will touch it :) ).